zml/tensor.zig

```diff
@@ -159,9 +159,33 @@ pub const Tensor = struct {
     }
 
     pub fn withSharding(self: Tensor, axes_: anytype) Tensor {
-        var res = self;
-        res._shape = self._shape.withSharding(axes_);
-        return res;
+        return switch (self._id) {
+            .mlir => {
```

Comment:
this looks wrong ? how did you tested it ?
Here you should emit the IR

Reply:
Good catch!